src/main/java/org/codebusters/audiogeek/spotifygazer/application/rest/getreleases/GetReleasesRestAdapter.java

```diff
@@ -0,0 +1,170 @@
+                                                                  @RequestParam(name = "page", defaultValue = "0") int page,
+                                                                  @RequestParam(name = "size", defaultValue = "${gazer.app.max-page-size}") int size) {
+        log.info("Processing POST request /get-current-releases: request={}, page={}, size={}", request, page, size);
+        checkPageable(page, size);
+        checkGenres(request.genres());
+        var albumsPage = query.getAlbums(new GetAlbumsByGenreFilter(request.genres(), PageRequest.of(page, size)));
+        var total = albumsPage.getTotalElements();
+        var albums = albumsPage.stream().map(this::convertToOutsideSchema).collect(toSet());
+        var response = GetReleasesResponse.builder()
+                .albums(albums)
+                .page(page)
+                .size(size)
+                .total(total)
+                .build();
+        log.debug("Successfully processed request: response={}", response);
+        return ok(response);
+    }
+
+    private void checkPageable(int page, int size) {
+        log.trace("Approving given pageable: page={} size={}", page, size);
+        checkPageSize(size);
+        checkPageIndex(page);
+        log.trace("Pageable is fully correct");
+    }
+
+    private void checkGenres(Set<String> genres) {
+        log.trace("Approving given genres: genres={}", genres);
+        checkIfGenresNull(genres);
+        checkGenresSize(genres.size());
+        genres.forEach(this::checkGenre);
+        log.trace("Genres are fully approved");
+    }
+
+    private void checkGenre(String genre) {
+        if (genre.length() > maxGenreLength) {
+            log.warn("Given genre is too long: genre={}", genre);
+            throw new GetReleasesApiException(GENRE_TOO_LONG, genre, maxGenreLength);
+        } else if (genre.length() == 0) {
+            log.warn("Given genre is empty");
+            throw new GetReleasesApiException(GENRE_EMPTY);
+        }
+        checkIfGenreContainsIllegalSymbol(genre);
+    }
+
+    private void checkIfGenresNull(Set<String> genres) {
+        if (genres == null) {
+            log.warn("Genres are null");
+            throw new GetReleasesApiException(GENRES_NULL);
+        }
+    }
+
+    private void checkGenresSize(int size) {
+        if (size == 0) {
+            log.warn("Genres are empty");
+            throw new GetReleasesApiException(GENRES_EMPTY);
+        } else if (size > maxGenresSize) {
+            log.warn("There is too many genres: size={}", size);
+            throw new GetReleasesApiException(GENRES_TOO_LONG, maxGenresSize);
+        }
+    }
+
+
+    private void checkIfGenreContainsIllegalSymbol(String genre) {
+        List<Character> chars = genre.chars().mapToObj(e -> (char) e).toList();
+        List<String> symbols = null;
+        if (charBlacklist.size() > 0 && charBlacklist.stream().anyMatch(genre::contains)) {
```

Comment:
Are you sure this is OK? The existing blacklist excludes whitelist configuration? What is business use case of this feature?